hq/markdown/snyk.md

```diff
@@ -0,0 +1,227 @@
+
+As Snyk is written in an interpreted script, this can be 'fixed' for now (if you use sbt > 1.0.0) using the following (mac) command:
+
+```
+sed -i '' 's/dependency-tree/dependencyTree/g' $(grep -wrl 'dependency-tree' /usr/local/lib/node_modules/snyk/node_modules/snyk-sbt-plugin/)
+```
+
+And to revert:
+```
+sed -i '' 's/dependencyTree/dependency-tree/g' $(grep -wrl 'dependencyTree' /usr/local/lib/node_modules/snyk/node_modules/snyk-sbt-plugin/)
+```
+
+#### With bugs in build.sbt
+
+If there is a bug in build.sbt, snyk is likely to get stuck on the `Querying vulnerabilities database...` stage.  Therefore
+it is suggested that you always invoke `sbt test` or similar first.  If  you see the `Querying vulnerabilities database...`
+step for more than ten seconds or so, you should check your `build.sbt` is good.
+
+## Use with Nodejs
+
+### Pre-requisites
+
+Ensure your node packages are up to date with the following:
+
+```
+rm node_modules
+npm install
+```
+
+### Invocation
+
+Scan for vulnerabilities with:
+
+```
+snyk test --file=package.json
+```
+
+Note that by default, Snyk will only scan for the dependencies found in the `dependencies` section.  It may be relevant
+to you to check for dependencies in development, which will only happen if you scan the `devDependencies` section using
+the `--dev` flag:
+
+```
+snyk test --file=package.json --dev
+```
+
+### Optional Magic
+
+Snyk provides a node-only `protect` command which will forcibly alter dependencies to remove vulnerabilities.  This
+requires setting up default locations for fetching libraries and rulesets for version changes, which is achieved using:
+
+```
+snyk wizard
+```
+
+And then answering questions.  Protection can then be requested using:
+
+```
+snyk protect
+```
+
+# Eliminating Vulnerabilities
+
+Clearly the best way to remove a vulnerability is to use a version of the library which does not have the vulnerability.
+This is usually achieved by upversioning (or in rarer cases downversioning).
+
+Alternatively, if a secure version of a library is not available, and an alternative is not found, it is possible to
+ignore the vulnerability temporarily.  This approach should be used with caution!
+
+## Upversioning
+
+Note that examples are all for sbt projects, however nodejs and other projects will be similar.
+
+When you visit a vulnerability report from the output (eg https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-31573),
+you may see a 'Remediation' section at the bottom.  If not, you will have to read the text and decide for yourself!
+
+If there is a known non-vulnerable version listed, you can simply change the dependency declaration as below:
+
+```
+$ git diff build.sbt
+diff --git a/build.sbt b/build.sbt
+index 434dbb2..d123098 100644
+--- a/build.sbt
++++ b/build.sbt
+@@ -29,7 +29,11 @@ libraryDependencies ++= Seq(
+   "org.typelevel" %% "cats" % "0.8.1",
+-  "my.library" %% "cleverstuff" % "0.0.99"
++  "my.library" %% "cleverstuff" % "1.0.0+"
```

Comment:
We try to use specific versions for our build definitions rather than ranges (e.g. `+`).

Reply:
OK.  I'll change the example.